### DISCUSSION TITLE
MQTT Publishing frequency

### DISCUSSION THREAD

**Discussion by silentrain23** [arendst/Tasmota#17592] [Ideas]
HI
I am trying sending a total of 10 readings in 1 single MQTT publish command every 10 seconds (I know I can change the tele_period to be less than 1, but my goal is different)
Any idea? I am working on Shelly 3EM 3phase and the xnrg_23_ade7880.ino and my goal is to have a MQTT JSON like this
`...
    "Voltage" : [ 220.3, 210.2, 215.3, v2, v2, v2, v3, v3, v3, v4, v4, v4, v5, v5, v5, v6, v6, v6, v7, v7, v7, v8, v8, v8, v8, v9, v9, v9 ],
    "Current" : [ 0.449, 0.013, 0.013, i2, i2, i2, i3, i3, i3, i4, i4, i4, i5, i5, i5, i6, i6, i6, i7, i7, i7, i8, i8, i8, i8, i9, i9, i9 ],
...
}` where v(n) and i(n) are the readings
Thank you

**Comment by sfromis** [comment]
Tasmota does not have features for doing that. What I'd do would only work on ESP32, where the Berry language is great for programming such local processing of data.

More generally, using `PowerDelta` you can greatly reduce the need for high-frequency reporting, by having extra messages triggered when consumption level is detected as going up or down by more than a set threshold.

**Reply by silentrain23** [reply]
I need to read a continuous flow to monitor a device. On the Shelly3Em I just have a 8266 chip with lower capabilities. I am trying to sample at a faster rate, store the result in an array, then publish via MQTT

**Reply by sfromis** [reply]
Yes, you do not need to explain again. That does not change my answer.

**Reply by barbudor** [reply]
You might be able to do that with Scripting-Language but I can't help you there

**Reply by silentrain23** [reply]
I am trying to sample the Energy.* every 200ms and send the results every 10 seconds to MQTT, but I cannot undestand how and when the Ade7880Read is called

**Reply by arendst** [reply]
The Tasmota ADE7880 implementation receives an interrupt every 100 cycles (being 1 second at 50Hz) allowing to read stable data. So shorter than 1 second is a no-go.

Every second the energy variables are updated waiting to be retrieved by either teleperiod, rules or GUI.

Your best bet is using rules to extract the information and provide it to you.

**Reply by silentrain23** [reply]
I am sorry I am new to this wonderful tool :( you mean using a ScriptLanguage by using a '>F' (100ms) that reads I, V, pf and stores the datas into an array. Then, every 10seconds the array is published to MQTT?

**Reply by barbudor** [reply]
> I am sorry I am new to this wonderful tool :( you mean using a ScriptLanguage by using a '>F' (100ms) that reads I, V, pf and stores the datas into an array. Then, every 10seconds the array is published to MQTT?

More or less
But as stated by Theo, the best frequency you would have is every second.
So better to use `>S` but I don't known if you can get values for voltage from `>S` or if only from `>T`

At some point, may be you don't have the right hardware or software platform for what you want to achieve...

**Reply by sfromis** [reply]
What Tasmota aims to do is energy consumption monitoring, not track fluctuations over short time intervals. Additionally, the hardware in cheap IoT devices cannot be expected to be up to such a task.

**Reply by silentrain23** [reply]
I succeded in reading the values every 100ms with the enrg[x] registers. Everything looks smooth. 
What  a great Platform you have done, guys!
